### PR TITLE
AggrTypeBuilder: mark aggregate as packed when field offset is not naturally aligned, to fix #4719

### LINKED ISSUE #4719: Incorrect LLVM type for struct with single align(1) member
This test case triggers the assert: https://github.com/ldc-developers/ldc/blob/48c1dff0466f5231619356ae33ef1a181888a1ff/ir/irtypeaggr.cpp#L43

```d
pragma(msg, TraceBuf.alignof); // prints 1
struct TraceBuf {
    align(1) uint args;
}

void foo() {
    byte[2] fixDescs;
    TraceBuf fixLog;

    auto dlg = delegate() {
        fixDescs[0] = 1;
        fixLog.args = 1;
    };
}
```

This is a minimized testcase, that no longer crashes LDC. The original case ran into the error
`Error: variable ... overlaps previous field. This is an ICE, please file an LDC issue.`

The reason is that the LLVM type is wrong since 1.31.0.

Used to be:
`%example.TraceBuf = type <{ i32 }>`
but now is:
`%example.TraceBuf = type { i32 }`, i.e. alignment is not 1 but 4.

### PR DESCRIPTION
**Pull request by JohanEngelen**
Alternative to https://github.com/ldc-developers/ldc/pull/4720

### CODE REVIEW

**Comment by kinke** [comment]
Excellent, I wasn't sure whether you'd expect me to fix this, glad I haven't started yet! :)

File: tests/codegen/gh4719.d [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// RUN: %ldc -c -output-ll -of=%t.ll %s
```
Comment (kinke) [inline tests/codegen/gh4719.d:1]: Are you planning on adding IR checks?

Comment (kinke) [reply tests/codegen/gh4719.d:1]: Or just a little speedup to skip the machine codegen?

Comment (JohanEngelen) [reply tests/codegen/gh4719.d:1]: Oh this is a good point, I think I should change it to do codegen (and possibly trigger bug further down the line)

Comment (kinke) [reply tests/codegen/gh4719.d:1]: [I've used `tests/compilable/` for pure does-it-compile tests without IR checks.]

**Review by kinke — APPROVED** [review]

**Pull request merged**